tests/test_search.py

```diff
@@ -151,14 +167,88 @@ def test_elasticsearch_exception(self):
+            url = self.build_url()
+            self.assertPostResponse(
+                path=url,
+                json_request_body=dict(es_query=smartseq2_paired_ends_query),
+                expected_code=requests.codes.internal_server_error,
+                expected_error=ExpectedErrorFields(code="internal_server_error",
+                                                   status=requests.codes.internal_server_error))
         finally:
             os.environ['DSS_ES_ENDPOINT'] = original_es_endpoint
             es_logger.setLevel(original_es_level)
 
+    def test_next_page_is_delivered_when_next_is_in_query(self):
+        bundles = self.populate_search_index(self.index_document, 150)
+        self.check_count(smartseq2_paired_ends_query, 150)
+        url = self.build_url()
+        search_obj = self.assertPostResponse(
+            path=url,
+            json_request_body=dict(es_query=smartseq2_paired_ends_query),
+            expected_code=requests.codes.ok)
+        found_bundles = search_obj.json['results']
+        self.verify_next_url(search_obj.response.headers['next'])
+        self.verify_search_result(search_obj.json, smartseq2_paired_ends_query, 100)
+        search_obj = self.assertPostResponse(
+            path=search_obj.response.headers['next'],
+            json_request_body=dict(es_query=smartseq2_paired_ends_query),
+            expected_code=requests.codes.ok)
+        found_bundles.extend(search_obj.json['results'])
+        self.verify_search_result(search_obj.json, smartseq2_paired_ends_query, 50)
+        self.verify_bundles(found_bundles, bundles)
+
+    def test_page_has_N_results_when_per_page_is_N(self):
+                        # per_page, expected
+        per_page_tests = [(10, 10),
+                          (100, 100),
+                          (500, 500)]  # max is 500
+
+        self.populate_search_index(self.index_document, 500)
+        self.check_count(smartseq2_paired_ends_query, 500)
+        for per_page, expected in per_page_tests:
+            url = self.build_url({"per_page": per_page})
+            with self.subTest(per_page=per_page, expected=expected):
+                search_obj = self.assertPostResponse(
+                    path=url,
+                    json_request_body=dict(es_query=smartseq2_paired_ends_query),
+                    expected_code=requests.codes.ok)
+                self.verify_search_result(search_obj.json, smartseq2_paired_ends_query, expected)
+
+    def test_error_returned_when_per_page_is_out_of_range(self):
+        expected_error = ExpectedErrorFields(code="illegal_arguments",
+                                             status=requests.codes.bad_request,
+                                             expect_stacktrace=True)
+        per_page_tests = [(9, {'expected_code': requests.codes.bad_request,
+                               'expected_error': expected_error}),  # min is 10
+                          (510, {'expected_code': requests.codes.bad_request,
```

Comment:
Did you intend to check with 501 as a bounds check?

Reply:
No I picked 510 intentionally. Do you think 501 would be a better test?

Reply:
If the valid range is 10-500, bounds checking is often/typically done with values one below (9, as you did) and one above (501). If in this case 510 is somehow a better test value that is fine, yet it would be good to add a comment to test describing why.

Reply:
it was arbitrary. I'll change it 501 do conform with general best testing practices.